src/components/blocks/index.tsx

```diff
@@ -0,0 +1,276 @@
+// copied from pages/trades.tsx
+
+import {
+  Text,
+  Box,
+  FormLabel,
+  NumberInput,
+  FormControl,
+  NumberInputField,
+} from "@chakra-ui/react";
+import { LoadingSpinner } from "@/components/util/loadingSpinner";
+import { useEffect, useRef, useState } from "react";
+import { BlockSummary } from "@/components/executionHistory/blockSummary";
+import { BlockInfo, LiquidityPositionEvent } from "@/utils/indexer/types/lps";
+import { SwapExecutionWithBlockHeight } from "@/utils/protos/types/DexQueryServiceClientInterface";
+import { BlockInfoMap, BlockSummaryMap } from "@/utils/types/block";
+
+export default function Blocks() {
+  // Go back hardcoded N blocks
+  const NUMBER_BLOCKS_IN_TIMELINE = 50;
+
+  const [isBlockRangeLoading, setIsBlockRangeLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isLineLoading, setIsLineLoading] = useState(true);
+
+  const [startingBlockHeight, setStartingBlockHeight] = useState(-1); // negative number meaning not set yet
+  const [endingBlockHeight, setEndingBlockHeight] = useState(-1); // negative number meaning not set yet
+  const [blockInfo, setBlockInfo] = useState<BlockInfoMap>({});
+  const [blockData, setBlockData] = useState<BlockSummaryMap>({});
+  const [userRequestedBlockEndHeight, setUserRequestedBlockEndHeight] =
+    useState(-1);
+
+  const currentStatusRef = useRef<HTMLDivElement>(null);
+  const originalStatusRef = useRef<HTMLDivElement>(null);
+  const [lineHeight, setLineHeight] = useState(0);
+  const [lineTop, setLineTop] = useState(0);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  // Get starting block number
+  useEffect(() => {
+    setIsBlockRangeLoading(true);
+    if (endingBlockHeight <= 0 || startingBlockHeight <= 0) {
+      var blockInfoPromise: Promise<BlockInfo[]>;
+      if (userRequestedBlockEndHeight >= 1) {
+        const startHeight = Math.max(
+          userRequestedBlockEndHeight - NUMBER_BLOCKS_IN_TIMELINE + 1,
+          1
+        ); // Lowest block_height is 1
+        blockInfoPromise = fetch(
+          `/api/blocks/${startHeight}/${userRequestedBlockEndHeight + 1}`
+        ).then((res) => res.json());
+      } else {
+        blockInfoPromise = fetch(
+          `/api/blocks/${NUMBER_BLOCKS_IN_TIMELINE}`
+        ).then((res) => res.json());
+      }
+      Promise.all([blockInfoPromise])
+        .then(([blockInfoResponse]) => {
+          const blockInfoList: BlockInfo[] = blockInfoResponse as BlockInfo[];
+          const blockInfoMap: BlockInfoMap = {};
+          blockInfoList.forEach((blockInfo: BlockInfo, i: number) => {
+            //console.log(blockInfo)
+            blockInfoMap[blockInfo["height"]] = blockInfo;
+          });
+
+          if (blockInfoList.length === 0) {
+            setIsLoading(false);
+            setError("No blocks found before: " + userRequestedBlockEndHeight);
+            console.log("No blocks found");
```

Comment:
suggestion/nitpick (non-blocking): remove `console.log`s. 

In the Penumbra Web repo, we have the `no-console` ESLint rule. Would be nice to have it here too